concept/type/ThingTypeImpl.java

```diff
@@ -28,275 +28,278 @@
+        thing_type_set_owns(((ConceptManagerImpl) transaction.concepts()).transaction, nativeObject, ((AttributeTypeImpl) attributeType).nativeObject, overriddenTypeNative, annotationsArray);
+    }
 
-        @Override
-        public final Stream<RoleTypeImpl> getPlaysExplicit() {
-            return stream(getPlaysExplicitReq(getLabel()))
-                    .flatMap(rp -> rp.getThingTypeGetPlaysExplicitResPart().getRoleTypesList().stream())
-                    .map(RoleTypeImpl::of);
-        }
+    @Override
+    public final Stream<RoleTypeImpl> getPlays(TypeDBTransaction transaction) {
+        return getPlays(transaction, Transitivity.Transitive);
+    }
 
-        @Override
-        public RoleTypeImpl getPlaysOverridden(RoleType roleType) {
-            ConceptProto.ThingType.GetPlaysOverridden.Res res = execute(
-                    getPlaysOverriddenReq(getLabel(), protoRoleType(roleType))
-            ).getThingTypeGetPlaysOverriddenRes();
-            switch (res.getResCase()) {
-                case ROLE_TYPE:
-                    return RoleTypeImpl.of(res.getRoleType());
-                default:
-                case RES_NOT_SET:
-                    return null;
-            }
-        }
+    @Override
+    public final Stream<RoleTypeImpl> getPlaysExplicit(TypeDBTransaction transaction) {
+        return getPlays(transaction, Transitivity.Explicit);
+    }
 
-        @Override
-        public Stream<AttributeTypeImpl> getOwns() {
-            return getOwns(emptySet());
-        }
+    private Stream<RoleTypeImpl> getPlays(TypeDBTransaction transaction, Transitivity transitivity) {
+        return thing_type_get_plays(((ConceptManagerImpl) transaction.concepts()).transaction, nativeObject, transitivity).stream().map(RoleTypeImpl::new);
+    }
 
-        @Override
-        public Stream<AttributeTypeImpl> getOwns(ValueType valueType) {
-            return getOwns(valueType, emptySet());
-        }
+    @Override
+    public RoleTypeImpl getPlaysOverridden(TypeDBTransaction transaction, RoleType roleType) {
+        com.vaticle.typedb.client.jni.Concept res = thing_type_get_plays_overridden(((ConceptManagerImpl) transaction.concepts()).transaction,
+                nativeObject, ((RoleTypeImpl) roleType).nativeObject);
+        if (res != null) return new RoleTypeImpl(res);
+        else return null;
+    }
 
-        @Override
-        public Stream<AttributeTypeImpl> getOwns(Set<TypeQLToken.Annotation> annotations) {
-            return stream(getOwnsReq(getLabel(), protoAnnotations(annotations)))
-                    .flatMap(rp -> rp.getThingTypeGetOwnsResPart().getAttributeTypesList().stream())
-                    .map(AttributeTypeImpl::of);
-        }
+    @Override
+    public Stream<AttributeTypeImpl> getOwns(TypeDBTransaction transaction) {
+        return getOwns(transaction, Transitivity.Transitive, set());
```

Comment:
use `emptySet()` and `emptyList()` when you use an empty immutable set/list to avoid unneeded allocations